gio-subclass/src/action.rs

```diff
@@ -0,0 +1,220 @@
+}
+
+unsafe extern "C" fn action_get_name<T: >
+(ptr: *mut GAction) -> *const c_char
+where
+    T::ImplType: ActionImpl<T>
+{
+    callback_guard!();
+    floating_reference_guard!(ptr);
+    let klass = &**(ptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(ffi::action_get_type())
+                                     as *const ActionStatic<T>;
+    let interface_static = &*(ptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    imp.get_name()
+}
+
+unsafe extern "C" fn action_get_parameter_type<T: >
+(ptr: *mut GAction) -> *const glib::GVariantType
+where
+    T::ImplType: ActionImpl<T>
+{
+    callback_guard!();
+    floating_reference_guard!(ptr);
+    let klass = &**(ptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(ffi::action_get_type())
+                                     as *const ActionStatic<T>;
+    let interface_static = &*(ptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    imp.get_parameter_type()
+}
+
+unsafe extern "C" fn action_get_state<T: >
+(ptr: *mut GAction) -> *mut glib::GVariant
+where
+    T::ImplType: ActionImpl<T>
+{
+    callback_guard!();
+    floating_reference_guard!(ptr);
+    let klass = &**(ptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(ffi::action_get_type())
+                                     as *const ActionStatic<T>;
+    let interface_static = &*(ptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    imp.get_state()
+}
+
+unsafe extern "C" fn action_get_state_hint<T: >
+(ptr: *mut GAction) -> *mut glib::GVariant
+where
+    T::ImplType: ActionImpl<T>
+{
+    callback_guard!();
+    floating_reference_guard!(ptr);
+    let klass = &**(ptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(ffi::action_get_type())
+                                     as *const ActionStatic<T>;
+    let interface_static = &*(ptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    imp.get_state_hint()
+}
+
+unsafe extern "C" fn action_get_state_type<T: >
+(ptr: *mut GAction) -> *const glib::GVariantType
+where
+    T::ImplType: ActionImpl<T>
+{
+    callback_guard!();
+    floating_reference_guard!(ptr);
+    let klass = &**(ptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(ffi::action_get_type())
+                                     as *const ActionStatic<T>;
+    let interface_static = &*(ptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    imp.get_state_type()
```

Comment:
Like signal trampolines, this needs all the conversion stuff for the return value (`from_glib` in this case) and also the parameters (none here). You also want to pass the `&gio::Action` via `&from_glib_borrow(ptr)` to this and all other functions